### DISCUSSION TITLE
Filesystem-based keymap management

### DISCUSSION THREAD

**Discussion by nickcoutsos** [nickcoutsos/keymap-editor#106] [Show and tell]
As a quick followup to https://github.com/nickcoutsos/keymap-editor/discussions/105 I've added another integration that leverages the [File System Access](https://developer.chrome.com/articles/file-system-access/) browser API.

This integration allows you to read and write directly to select `.keymap` files on your computer to cut down on the tedium of switching programs and pasting changes from your clipboard.

Some notes:

- This API is currently only supported in Chromium-based browsers, so Firefox and Safari users won't see it as an option in the _Sources_ dropdown.
- If your keymap can be matched (by filename) to a known ZMK keyboard it will select a pre-generated layout. You can't (currently) provide an alternate layout if one isn't available or if your filename doesn't match correctly.
- ~~Right now you need to re-select your keymap file(s) whenever you load the page. I still need to look at using indexdb to persist file handles between sessions.~~ your file selection is persisted between page loads, but the browser API requires that you need to grant permissions each time.

Let me know what you think!

**Comment by aexvir** [comment]
love this feature, I use it as an editor for my keymap file instead of opening it as text
